service/history/workflow/mutable_state_impl.go

```diff
@@ -4304,10 +4305,6 @@ func (ms *MutableStateImpl) updateWithLastWriteEvent(
 	return nil
 }
 
-func (ms *MutableStateImpl) canReplicateEvents() bool {
-	return ms.namespaceEntry.ReplicationPolicy() == namespace.ReplicationPolicyMultiCluster
```

Comment:
this assumption is broken

Reply:
Can we just reuse this one

Reply:
what is the difference?

Reply:
I think the name canReplicateEvents make sense as it return a boolean. But it is up to you.